tests/unit/Type/ClassStringTypeTest.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Tests\Unit\Type;
+
+use Psl\Collection;
+use Psl\Type;
+
+final class ClassStringTypeTest extends TypeTest
+{
+    public function getType(): Type\TypeInterface
+    {
+        return Type\class_string(Collection\CollectionInterface::class);
+    }
+
+    public function getValidCoercions(): iterable
+    {
+        yield [$_ = Collection\Vector::class, $_];
+        yield [$_ = Collection\MutableVector::class, $_];
+        yield [$_ = Collection\Map::class, $_];
+        yield [$_ = Collection\MutableMap::class, $_];
```

Comment:
For completeness : could you add a valid coercion that checks for subclasses of a given class / interface type?

It behaves like this on psalm and given your code I think it behaves the same. But just to be sure here.

See inheritance cases in:
https://psalm.dev/r/03a91eba69

Reply:
I don't understand :-( `Psl\Tests\Unit\Type\TypeTest` only allows to pass one value from the `getType` and test against it.

Reply:
Oh yeah indeed. I was assuming you could also vary in `getType`, but that's not how it works.
Thanks for pointing it out. It seems sufficient this way.